system/index.js

```diff
@@ -30,54 +48,77 @@ module.exports = async (message) => {
   const payload = message.payload;
   // Remap headers to a standard JS object
   const headers = message.headers.toRiffHeaders();
-  Object.keys(headers).map((key) => {headers[key] = message.headers.getValue(key)});
-  if (DEBUG) {
-    console.log('==System Function Start==');
-    console.log(`HEADERS: ${JSON.stringify(headers)}`);
-    console.log(`PAYLOAD: ${JSON.stringify(payload)}`);
-    console.log(`MIDDLEWARE_FUNCTION_URI: ${MIDDLEWARE_FUNCTION_URI}`);
-    console.log('==Middleware Function(s) Start==');
-  }
+  Object.keys(headers).map((key) => { headers[key] = message.headers.getValue(key) });
+  const requestID = headers['x-request-id']
+
+  const logLevel = DEBUG ? 'debug' : 'info';
+  const log = initLogging(logLevel, requestID);
+
+  log.debug({
+    headers: JSON.stringify(headers),
+    payload: JSON.stringify(payload),
+    middware_function_uri: MIDDLEWARE_FUNCTION_URI,
+    status: 'system function start',
+  });
+
+  log.debug({ status: 'middleware function(s) start' });
 
   const state = {};
   let middlewareResult = [payload];
 
   await Promise.all(middlewareFns.map(async (middleware) => {
-        try {
-          // input should be immutable
-          const input = {
-            payload: typeof payload == "object" ? Object.assign({}, payload) : payload,
-            headers: Object.assign({}, headers)
-          };
-          if (DEBUG) {
-            console.log(`MIDDLEWARE INPUT: ${JSON.stringify(input)}`);
-            console.log(`MIDDLEWARE STATE: ${JSON.stringify(state)}`);
-            console.log(`MIDDLEWARE RESULT: ${JSON.stringify(middlewareResult)}`);
-          }
-          middlewareResult = await middleware(input, state, middlewareResult);
-          if (DEBUG) {
-            console.log(`MIDDLEWARE RETURNED: ${JSON.stringify(middlewareResult)}`);
-          }
-          if (!Array.isArray(middlewareResult)) {
-            throw new Error('Invalid return type, middleware must return an array of arguments')
-          }
-        } catch (err) {
-          throw err
-        }
-      })
-  );
-
-  if (DEBUG) {
-    console.log('==Middleware Function(s) End==');
-    console.log(`USER FUNCTION RECEIVED ARGS: ${JSON.stringify(middlewareResult)}`);
-  }
+    try {
+      // input should be immutable
+      const input = {
+        payload: typeof payload == "object" ? Object.assign({}, payload) : payload,
+        headers: Object.assign({}, headers)
+      };
+
+      log.debug({
+        middleware_input: JSON.stringify(input),
```

Comment:
Did we ever decide if we wanted to standardize on snake or camel case?

`middleware_input`
`middleware_state`
`middleware_result`

Reply:
snake_case it is